perfkitbenchmarker/scripts/database_scripts/plot_scatter_points.py

```diff
@@ -0,0 +1,125 @@
+
+# This variable controls the verticle lines inside the chart.
+#   0 means no vertical lines.
+Y_TICS = '100'
+ITERATIONS = '10'
+
+
+class GnuplotInfo():
+
+
+  def __init__(self, gnuplot_data_filename,
+               entries_per_run,
+               run_uri,
+               y_max,
+               title=None):
+    """Initialize GnuplotInfo object.
+    Args:
+      gnuplot_data_filename: filename of TPS data.
+      entries_per_run: Number of TPS values collected for each run.
+      run_uri: (string) run identifier.
+      y_max: maximum y value. Used for y-axis limit.
+      title: (optional, string) Chart title.
+    """
+    self.gnuplot_data_filename = gnuplot_data_filename
+    self._generate_filenames(run_uri)
+    self.X_INTERVAL = str(entries_per_run)
+    self.Y_HEIGHT = str(int(100 * round(float(y_max) / 100)))
+    self.title = title
+
+  def _generate_filenames(self, run_uri):
+    """Sets filename (with path) of gnuplot input and chart.
+    Args:
+      run_uri: (string) run identifier.
+    """
+    date_string = DATETIME_FORMAT.format(datetime.datetime.now())
+    identifier = date_string + run_uri + '_innodb_pages.png'
+    self.output_chart = os.path.join(
+        os.path.dirname(__file__), '..', '..', '..', 'charts',
+        identifier)
+    self.output_gnuplot_file = self.output_chart + '_gnuplot_input'
+
+  def create_file(self):
+    """Generates a gnuplot info file.
+    Returns:
+      output_file (string): Name of gnuplot output file.
+      output_chart (string): Name of output chart file.
+    """
+    color = '38761d'
+
+    # Titles for the data series
+    title = self.title or '1TB DB Instance'
+
+    output_file = open(self.output_gnuplot_file, 'w')
+    output_file.write('set terminal pngcairo size 1500,800 '
+                      'enhanced font "Verdana,12"\n')
+    output_file.write('set output "' + self.output_chart + '"\n')
+    output_file.write('set multiplot\n')
+    output_file.write('set grid\n')
+    output_file.write('set border 4095 ls 0 lc rgb \"black\"\n')
+    output_file.write('set title (\"' + CHART_TITLE +
+                      '") font \"aerial, 14\"\n')
+    output_file.write('set xlabel "' + X_LABEL + '"\n')
+    output_file.write('set ylabel "' + Y_LABEL + '"\n')
+
+    # If you want to move the legend to the top left, use this:
+    output_file.write('set key left top\n')
+
+    if self.Y_HEIGHT > 0:
+      output_file.write('y=' + self.Y_HEIGHT + '\n')
+      output_file.write('set yrange [0:y]\n')
+      output_file.write('set ytics ' + Y_TICS + '\n')
+      output_file.write('unset xtics\n')
+
+      output_file.write('thread=1\n')
+      output_file.write('x=0\n')
+      output_file.write('do for [t=1:' + ITERATIONS + ':+1] {\n')
+      output_file.write(
+          '\tset label (sprintf(\"%d\", thread)) at x+20, 0  offset -2\n')
+      output_file.write(
+          '\tset arrow from x,0 to x,y nohead ls 0 lc rgb \"blue\"\n')
+      output_file.write('\tthread=thread*2\n')
```

Comment:
put a TODO here to indicate that in this code here we assume the run configuration is a step load run with each step have 2x more threads than previous step. Future work should take a list of thread counts and then plot them properly here.